### ISSUE TITLE
Source creating multiple parallel consumers

### ISSUE THREAD

**Issue by mmatloka** [akka/reactive-kafka#231]
According to some Kafka training materials creating multiple parallel consumers may be beneficial on stronger machines (if there are more partitions than consumers). It might be interesting to have `Source` creating multiple threads with multiple `KafkaConsumer`s. From API perspective it could be similar to handling `committablePartitionedSource`.

**Comment by 13h3r** [comment]
You always can manually create any number of `Source`s and merge them into a one big source. Each of this sources will be scheduled independently and run in parallel.

**Comment by graphex** [comment]
I think there are a couple of things to watch out for when doing this. I tried merging multiple committableSources a while ago, and the first thing i ran in to was an MBean collision warning from the Kafka client itself. Not a showstopper but certainly annoying.

Then, if memory serves, when trying to commit, there was an actor lookup going behind the scenes in reactive kafka that seemed to operate on group ID as a lookup key (unfortunately not client ID which i could have differentiated) and commits failed when multiple sources were consuming as the same consumer group and a committable batch was made from a merged source of them.

I'm not sure if the same behavior is still present in the .11 release version but I certainly ran in to some problems with earlier versions creating multiple sources with the same settings, and had to move to committablePartitionedSources. Would be worth a repro and a git issue if the behavior is still there. I had meant to report this earlier but just haven't had time to get it packaged up cohesively.

Sean

**Comment by dbcfd** [comment]
Is this issue still relevant? I currently run something like 

`
val clientSettings = settings.withClientId(parallelClientId)
val src = Consumer.committablePartitionedSource[Array[Byte], Array[Byte]](clientSettings, Subscriptions.topicPattern(wlRegex)
src.flatMapMerge(config.maxPartitionsToHandle, tpSrc => {
          tpSrc.groupedWithin(config.commit.groupSize, config.commit.groupWithin)
            .filter(_.nonEmpty)
        })`

and have no issues committing.

**Comment by nishkamravi2** [comment]
@dbcdf I don't think that would create multiple parallel consumers under the hood.

**Comment by nishkamravi2** [comment]
@graphex Mind sharing your efforts around running multiple parallel consumers?

**Comment by 2m** [comment]
This can be achieved by having multiple sources and merging them to one stream. Closing this ticket for now. Feel free to open a new ticket if you find problems with that strategy.
